src/NServiceBus.Core/Unicast/Behaviors/MessageHandler.cs

```diff
@@ -7,14 +7,75 @@
     /// </summary>
     public class MessageHandler
     {
+        Action<object, object, object> invocationWithContext;
+
+        /// <summary>
+        /// Creates a new instance of the message handler
+        /// </summary>
+        [ObsoleteEx(TreatAsErrorFromVersion = "6", RemoveInVersion = "7", ReplacementTypeOrMember = "MessageHandler(Action<object, object, object> invocationWithContext, Type handlerType)")]
```

Comment:
you cant have an obsolete on a public API that points to a internal member as the replacement